tests/unittests/InterpreterTest.cpp

```diff
@@ -896,3 +896,84 @@ TEST(Interpreter, nonLinearClassifier) {
     EXPECT_NEAR(RH.at({0, std::get<2>(tests[i])}), 1.0, 0.2);
   }
 }
+
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples) {
+  auto L = labels.getHandle<size_t>();
+  auto Im = images.getHandle<>();
+  images.zero();
+
+  for (size_t i = 0; i < numSamples; i++) {
+    L.at({i, 0}) = i % 2;
+    size_t target = rand() % 30 + 1;
+    for (size_t x = 0; x < 32; x++) {
+      for (size_t y = 0; y < 32; y++) {
+        if (i % 2 == 0) {
+          /// line
+          if (x == target)
+            Im.at({i, x, y, 0u}) = 1;
+        } else {
+          /// cross
+          if (x == target || y == target)
+            Im.at({i, x, y, 0u}) = 1;
+        }
+      }
+    }
+  }
+  printf("\n");
+}
+
+TEST(Interpreter, simpleConv) {
+  unsigned numSamples = 50;
+  ExecutionEngine EE;
+
+  EE.getConfig().learningRate = 0.001;
+  EE.getConfig().momentum = 0.9;
+
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+  F->setName("simpleConv");
+
+  auto *input = mod.createVariable(ElemKind::FloatTy, {numSamples, 32, 32, 1}, "input",
+                                   VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *ex = mod.createVariable(ElemKind::IndexTy, {numSamples, 1}, "exp",
+                                VisibilityKind::Public, Variable::TrainKind::None);
+
+  auto *CV0 = F->createConv("conv1", input, 6, 5, 1, 2, 1);
+  auto *AP0 = F->createPoolAvg("pool1", CV0, 2, 2, 0);
+  auto *CV1 = F->createConv("conv2", AP0, 6, 5, 1, 2, 1);
+  auto *AP1 = F->createPoolAvg("pool2", CV1, 2, 2, 0);
+  auto *TANH1 = F->createTanh("tanh2", AP1);  
+  auto *FCL1 = F->createFullyConnected("fc", TANH1, 120);
+  auto *TANH2 = F->createTanh("tanh3", FCL1);
+  auto *FCL2 = F->createFullyConnected("fc1", TANH2, 2);
+  auto *SM = F->createSoftMax("sm", FCL2, ex);                                                                                                                                      
```

Comment:
How many labels are you generating here?

Can you use F->dumpDAG() and check the shape of the output?